tests/Geocoder/Tests/TestCase.php

```diff
@@ -27,19 +28,16 @@ protected function getMockAdapter($expects = null)
             ->method('__toString')
             ->will($this->returnValue(''));
 
-        $response = $this->getMock('Psr\Http\Message\MessageInterface');
+        $response = $this->getMock('Psr\Http\Message\ResponseInterface');
```

Comment:
I would recommend using actual implementations for testing instead. This is what we are going to do in HTTPlug libraries as well. The reasoning behind is that HTTP Messages can be interpreted as ValueObjects/DTOs as well and as such, they shouldn't be mocked. The actual behaviour is not important, rather the value they carry.